lib/storage/limits/limits.go

```diff
@@ -0,0 +1,113 @@
+package limits
+
+import (
+	"flag"
+	"sync/atomic"
+	"time"
+
+	"github.com/VictoriaMetrics/metrics"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/prompbmarshal"
+)
+
+const maxLabelNameLen = 256
+
+var (
+	maxLabelsPerTimeseries = flag.Int("maxLabelsPerTimeseries", 30, "The maximum number of labels per time series to be accepted. Timeseries with superfluous labels are dropped. In this case the vm_series_dropped_total{reason=\"too_many_labels\"} metric at /metrics page is incremented")
+	maxLabelValueLen       = flag.Int("maxLabelValueLen", 4*1024, "The maximum length of label values in the accepted time series. Metrics with longer label value are dropped. In this case the vm_series_dropped_total{reason=\"too_long_label_value\"} metric at /metrics page is incremented")
+)
+
+var (
+	droppedSeriesWithTooManyLabelsLogTicker     = time.NewTicker(5 * time.Second)
+	droppedSeriesWithTooLongLabelNameLogTicker  = time.NewTicker(5 * time.Second)
+	droppedSeriesWithTooLongLabelValueLogTicker = time.NewTicker(5 * time.Second)
+)
+
+var (
+	// droppedSeriesWithTooManyLabels is the number of dropped series with too many labels
+	droppedSeriesWithTooManyLabels atomic.Uint64
+
+	// droppedSeriesWithTooLongLabelName is the number of dropped series which contain labels with too long names
+	droppedSeriesWithTooLongLabelName atomic.Uint64
+
+	// droppedSeriesWithTooLongLabelValue is the number of dropped series which contain labels with too long values
+	droppedSeriesWithTooLongLabelValue atomic.Uint64
+)
+
+var (
+	_ = metrics.NewGauge(`vm_series_dropped_total{reason="too_many_labels"}`, func() float64 {
```

Comment:
wdyt about using `vm_rows_ignored_total` name instead to remain consistent with other similar metrics exposed by vmstorage?

Reply:
updated